eip/EIPS/eip-4337.md

```diff
@@ -58,45 +69,88 @@ The core interface of the entry point contract is as follows:
+* MUST validate the signature is a valid signature of the `requestId`
+* MUST pay the entryPoint (caller) at least the "missingWalletFunds" (which might be zero, in case current wallet's deposit is high enough)
+* the wallet MAY pay more than this minimum, to cover future transactions
+* Note that in case the request comes through an aggregator, the signature will be empty.
+
+A Wallet that supports aggregated signature should have the interface:
+```solidity
+interface IAggregatedWallet is IWallet {
+  function getAggregator() view returns (address);
+}
+```
+* **getAggregator()** returns the aggregator this wallet supports.
+* The wallet is guaranteed to that the EntryPoint will only call its `validateUserOp` (with no signature) after calling the aggregator's validateSignatures
+* The wallet should also support aggregator-specific getter (e.g. `getAggregationInfo()`).
+  This method should export the wallet's public-key to the aggregator, and possibly more info
+* validateUserOp shouldn't validate the signature (since it is verified by the aggregator),
+  but it MUST verify the signature field is shorter than 63 bytes
+
+The core interface required by an aggregator is:
+```solidity
+interface IAggregator {
+
+  function validateSignatures(address entryPoint, UserOperation[] calldata userOps, bytes calldata signature) view external;
+}
 ```
 
+* bundlers/nodes must have explicit support for each aggregator, and know the code they need to calculate its aggregated signature.
+* the validateSignatures method is called by the entrypoint to let the aggregator validate the aggregate signature is valid for all UserOps.
+* if the signature is invalid or doesn't match, the aggregator should revert.
+* the aggregator SHOULDN'T call the getAggregator() of each wallet: the entrypoint already validated that on each userOp before calling validateSignatures
+
 ### Required entry point contract functionality
 
 The entry point's `handleOps` function must perform the following steps (we first describe the simpler non-paymaster case). It must make two loops, the **verification loop** and the **execution loop**. In the verification loop, the `handleOps` call must perform the following steps for each `UserOperation`:
-
+* Before the first validation loop, if an aggregator is specified, the entryPoint must call its `validateSignatures(UserOperation[], aggregatedSignature)`, and revert if the validation fails
 * **Create the wallet if it does not yet exist**, using the initcode provided in the `UserOperation`. If the wallet does not exist, _and_ the initcode is empty, or the newly deployed contract address differs from `UserOperation.sender`, the call must fail.
+* in case the UserOperation contains an empty signature, the entryPoint MUST call the wallet's getAggregator, and MUST revert in case the wallet doesn't implement getAggregator or it doesn't match the given **aggregator** parameter
```

Comment:
As I have said above, I consider reverting this so that the wallet checks the aggregator address internally instead of trusting that the EntryPoint did it for him. Wallet should trust the EntryPoint not to lie about the used 'aggregator' address.